community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -1382,6 +1388,14 @@ private StatementOperationParts buildStatementOperations(
         return parts;
     }
 
+    public void maintenance()
```

Comment:
This method is only used in `IdBufferingRoleSwitchIT` and is slightly confusing for a datasource. What do you think about putting `bufferingIdGeneratorFactory` dependency resolver instead?